include/seqan/journaled_string_tree/journaled_string_tree_traverser.h

```diff
@@ -46,33 +46,33 @@ namespace seqan
 // Tags, Classes, Enums
 // ============================================================================
 
-template <typename TJst, typename TSpec, typename TObserverList>
-class TraverserImpl<TJst, JstTraversalSpec<TSpec>, TObserverList> : public Observable<TObserverList>
+template <typename TJst, typename TSpec>
+class TraverserImpl<TJst, JstTraversalSpec<TSpec> >
 {
 public:
-    typedef Observable<TObserverList>                                   TSuper;
     typedef typename Member<TraverserImpl, TraverserStackMember>::Type  TStack;
     typedef typename Value<TStack>::Type                                TNode;
     typedef typename Size<TraverserImpl>::Type                          TSize;
     typedef std::shared_ptr<TStack>                                     TStackPtr;
     typedef JstBuffer_<TJst>                                            TBuffer;  // Provides a sequence context.
     typedef std::shared_ptr<TBuffer>                                    TBufferPtr;
-
-
+    typedef typename Member<TJst, JstDeltaMapMember>::Type              TDeltaMap;
+    typedef typename DeltaCoverage<TDeltaMap>::Type                     TCoverage;
 
     // __ Member Variables ____________________________________________________
 
-    TJst *      _contPtr;  // TODO(rmaerker): Should always be const.
-    TSize       _branchLength   = 1;
+    TJst *      _contPtr;  // TODO(rmaerker): Underlying container should be const.
     TSize       _contextSize    = 1;
+    TSize       _branchLength   = 1;
     TStackPtr   _stackPtr;
     TBufferPtr  _bufferPtr;
+    TCoverage   _baseCov;
+    bool        _needInitialization = true;
 
```

Comment:
does any of these actually contain data? If yes, we should now create move-constructors and move-assignment operators, see http://en.cppreference.com/w/cpp/language/rule_of_three
Especially see the note on replacing both assignment operators with one that fulfills both requirements.